src/Symfony/Component/AssetMapper/Compiler/JavaScriptImportPathCompiler.php

```diff
@@ -141,6 +141,16 @@ private function isCommentedOut(mixed $offsetStart, string $fullContent): bool
         return false;
     }
 
+    private function isEnclosedInQuotes(mixed $offsetStart, string $fullContent): bool
+    {
+        $lineStart = strrpos($fullContent, "\n", $offsetStart - \strlen($fullContent));
+        // Remove enclosed strings before the import
+        $lineContentBeforeImport = substr($fullContent, $lineStart, $offsetStart - $lineStart);
+        $lineWithoutStrings = preg_replace('/\'(?:[^\'\\\\]|\\\\.)*\'|"(?:[^"\\\\]|\\\\.)*"/', '', $lineContentBeforeImport);
```

Comment:
[template literals (`)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are quite common in modern js, and this syntax allows multiline strings making this check quite a challenge

Reply:
@kaznovac this PR fixes a bug with console log / deprecation / help messages (cf the original issue)

Handling multi-line strings/comments can reveal itself really complex, and probably  would need there some concrete JS parser. 

I must admit i doubt any library would use multi-line template comments in source code containing import instructions.. but if/when that happen we could handle it in a later PR ?

WDYT ?

Reply:
alternative implementation idea: we merge this regexp with the one that checks for imports.
Since regexp are greedy, this will make the filter in one go, with just a quick check at the start of the closure

Reply:
@nicolas-grekas 

Tried your idea but..  that goes beyond my regexp competences.... :) 

--> https://regex101.com/r/zug2ot/2

In the following case, we need to ignore foo.js import.. but want to extract the bar.js import

```
console.log('import {myFunction} from "foo.js"'); import {myFunction} from "bar.js";
```

Reply:
@nicolas-grekas got it, and it seems to work perfectly ! Thank you :) 

See https://regex101.com/r/qFoeoR/1

Reply:
@smnandre great work && sorry for the late answer

p.s. please note that this method can now be removed